aten/src/ATen/Dispatch.h

```diff
@@ -119,26 +138,6 @@ inline void deprecated_AT_DISPATCH_ALL_TYPES_AND_HALF_AND_COMPLEX() {}
     }                                                                        \
   }()
 
-#define AT_DISPATCH_ALL_TYPES_AND_HALF(TYPE, NAME, ...)                      \
```

Comment:
why are you deleting AT_DISPATCH_ALL_TYPES_AND_HALF?  We want to maintain this for backwards compatibility, see https://github.com/pytorch/pytorch/pull/17996.

Reply:
I didn't know about the backwards compatibility issue. I deleted it as we decided to replace some of these macros with  AT_DISPATCH_ALL_TYPES_AND(SCALARTYPE, TYPE, NAME, ...).
Reverting.